contracts/Proxy/UpgradableProxy.sol

```diff
@@ -0,0 +1,36 @@
+// SPDX-License-Identifier: GPL-3.0
+
+pragma solidity ^0.6.9;
+
+import "../Interface/IImplementationAuthority.sol";
+
+contract Proxy {
+    address implementationAuthority;
+
+    constructor(address _implementationAuthority) public {
+        // save the code address
+        implementationAuthority = _implementationAuthority;
+        address contractLogic = IImplementationAuthority(_implementationAuthority).getImplementation();
+        assembly { // solium-disable-line
+            sstore(0xc5f16f0fcc639fa48a6947836d9850f504798523bf8c9a3a87d5876cf622bcf7, contractLogic)
```

Comment:
You should white a comment that explains what "0xc5f16f0fcc639fa48a6947836d9850f504798523bf8c9a3a87d5876cf622bcf7" means for.

Also, you store the contractLogic but you do not use the stored variable anywhere.

Reply:
Besides, if the implementation is updated, this state variable is never updated, so this state variable is doesn't give valuable information.
Either remove it, or make it updated whenever the implementation used is updated.

Reply:
> You should white a comment that explains what "0xc5f16f0fcc639fa48a6947836d9850f504798523bf8c9a3a87d5876cf622bcf7" means for.
> 
> Also, you store the contractLogic but you do not use the stored variable anywhere.

I'm using it on the fallback

Reply:
No, you don't?

Reply:
```solidity
address logic = IImplementationAuthority(implementationAuthority).getImplementation();

        assembly { // solium-disable-line
        calldatacopy(0x0, 0x0, calldatasize())
        let success := delegatecall(sub(gas(), 10000), logic, 0x0, calldatasize(), 0, 0)
```

I don't see `0xc5f16f0fcc639fa48a6947836d9850f504798523bf8c9a3a87d5876cf622bcf7` here.

Reply:
the logic is indeed here, it is not necessary to reuse the location knowing that the Logic Contract is stored at the defined storage position, preventing the collision between variables in proxy and logical contracts

+ Proxiable is not used anymore we point directly to the address of the implementation

how would you do it?

Reply:
It make no sense to store the contractLogic with sstore if you retrieve the logic address with `getImplementation` each time you need it.
The stored value at `0xc5f16f0fcc639fa48a6947836d9850f504798523bf8c9a3a87d5876cf622bcf7` is never read, and when implementation is updated, its value remains to the first implementation so it is just a useless variable.

Just remove it.

Reply:
I misread the message, I thought you were referring to the variable state, yes sure since we removed Proxiable. Done